tests/trainer/test_trainer.py

```diff
@@ -1179,3 +1180,22 @@ def test_row_log_interval(log_metrics_mock, tmpdir, train_batches, max_steps, lo
     trainer.fit(model)
     expected_calls = [call(metrics=ANY, step=s) for s in range(log_interval - 1, max_steps, log_interval)]
     log_metrics_mock.assert_has_calls(expected_calls)
+
+    
+@pytest.mark.parametrize(['input_arg', 'expected'], [
+    (None, PassThroughProfiler),
+    (SimpleProfiler(), SimpleProfiler),
+    (AdvancedProfiler(), AdvancedProfiler),
+    ('simple', SimpleProfiler),
+    ('advanced', AdvancedProfiler),
+])
+def test_trainer_profiler_correct_args(input_arg, expected):
+    kwargs = {'profiler': input_arg} if input_arg is not None else {}
+    trainer = Trainer(**kwargs)
```

Comment:
```suggestion
@pytest.mark.parametrize(['profiler', 'expected'], [
    (None, PassThroughProfiler),
    (SimpleProfiler(), SimpleProfiler),
    (AdvancedProfiler(), AdvancedProfiler),
    ('simple', SimpleProfiler),
    ('advanced', AdvancedProfiler),
])
def test_trainer_profiler_correct_args(profiler, expected):
    trainer = Trainer(profiler=profiler)
```

Reply:
Thanks for the suggestion, I kept the `kwargs` since this way it also checks if `None` is used as default if nothing is supplied as the `profiler` arg.